### PR TITLE
Add combined author/mitid during indexing

### LINKED ISSUE #42: Add combined author/mit id field to solr during summary
Changes in the summary script depend on a string field with `author:mitid`. This is in addition to the two separate `author` and `mitid` fields.

### PR DESCRIPTION
**Pull request by gravesm**
Closes #42.

### CODE REVIEW

File: tests/unit/test_summary.py [hunk 1]
```diff
@@ -214,9 +215,17 @@ def test_dictify_converts_facet_count_to_dictionary():
...
 def test_split_author_returns_none_for_invalid_author():
     assert split_author(':Foo Bar') is None
     assert split_author('Foo Bar') is None
+
+
+def test_join_author_joins_author_parts():
+    assert join_author({'name': 'Cat, Lucy', 'mitid': '1234'}) == \
+        '1234:Cat, Lucy'
+
+def test_join_author_uses_empty_string_for_missing_items():
```
Comment (JPrevost) [inline tests/unit/test_summary.py]: Should there also be a test for a missing name and not just a missing ID?

**Comment by JPrevost** [comment]
:+1:

File: tests/unit/test_summary.py (hunk 1 above)
Comment (gravesm) [inline tests/unit/test_summary.py]: In the spirit of completeness, probably, but the data should never be such that we have an ID but no name, or neither a name nor an ID.

Comment (JPrevost) [inline tests/unit/test_summary.py]: Works for me.

**Pull request merged**